src/CachePlugin.php

```diff
@@ -46,19 +54,66 @@
      *              we have to store the cache for a longer time than the server originally says it is valid for.
      *              We store a cache item for $cache_lifetime + max age of the response.
      *     @var array $methods list of request methods which can be cached.
+     *     @var array $respect_response_cache_directives list of cache directives this plugin will respect while caching responses.
      * }
      */
     public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
     {
         $this->pool = $pool;
         $this->streamFactory = $streamFactory;
 
+        if (isset($config['respect_cache_headers']) && $config['respect_response_cache_directives']) {
```

Comment:
the second check also needs to have an isset. as is, this will be a fatal error if respect_response_cache_directives is not set. (and throw no exception if respect_cache_headers is set and the other is `[]`)